cmd/tnf/claim/show/failures/failures.go

```diff
@@ -0,0 +1,304 @@
+	for _, ts := range testSuites {
+		fmt.Printf("Test Suite: %s\n", ts.TestSuiteName)
+		for _, tc := range ts.FailingTestCases {
+			fmt.Printf("  Test Case: %s\n", tc.TestCaseName)
+			fmt.Printf("    Description: %s\n", tc.TestCaseDescription)
+			fmt.Printf("    Failure reasons:\n")
+			for i := range tc.NonCompliantObjects {
+				nonCompliantObject := tc.NonCompliantObjects[i]
+				fmt.Printf("      %2d - Type: %s, Reason: %s\n", i+1, nonCompliantObject.Type, nonCompliantObject.Reason)
+				fmt.Printf("           ")
+				for i := range nonCompliantObject.Spec.fields {
+					if i != 0 {
+						fmt.Printf(", ")
+					}
+					field := nonCompliantObject.Spec.fields[i]
+					fmt.Printf("%s: %s", field.key, field.value)
+				}
+				fmt.Printf("\n")
+			}
+		}
+	}
+}
+
+// Prints the failures in json format.
+func printFailuresJSON(testSuites []FailedTestSuite) {
+	type ClaimFailures struct {
+		Failures []FailedTestSuite `json:"testSuites"`
+	}
+
+	claimFailures := ClaimFailures{Failures: testSuites}
+	bytes, err := json.MarshalIndent(claimFailures, "", "  ")
+	if err != nil {
+		log.Fatalf("failed to marshal failures: %v", err)
+	}
+
+	fmt.Printf("%s\n", string(bytes))
+}
+
+// Creates a list of FailingTestSuite from the results parsed from a claim file. The parsed
+// results in claimResultsByTestSuite var maps a test suite name to a list of TestCaseResult,
+// which are processed to create the list of FailingTestSuite, filtering out those test suites
+// that don't exist in the targetTestSuites map.
+func getFailedTestCasesByTestSuite(claimResultsByTestSuite map[string][]*claim.TestCaseResult, targetTestSuites map[string]bool) ([]FailedTestSuite, error) {
+	testSuites := []FailedTestSuite{}
+	for testSuite := range claimResultsByTestSuite {
+		if targetTestSuites != nil && !targetTestSuites[testSuite] {
+			continue
+		}
+
+		failedTcs := []FailedTestCase{}
+		for _, tc := range claimResultsByTestSuite[testSuite] {
+			if tc.State != "failed" {
+				continue
+			}
+
+			nonCompliantObjects, err := getNonCompliantObjectsFromFailureReason(tc.FailureReason)
+			if err != nil {
+				return nil, fmt.Errorf("test suite %s, test case %s : failed to parse non compliant objects: %v", testSuite, tc.TestID.ID, err)
+			}
+
+			failingTc := FailedTestCase{
+				TestCaseName:        tc.TestID.ID,
+				TestCaseDescription: tc.Description,
+				NonCompliantObjects: nonCompliantObjects,
+			}
+
+			failedTcs = append(failedTcs, failingTc)
+		}
+
+		if len(failedTcs) > 0 {
+			testSuites = append(testSuites, FailedTestSuite{
+				TestSuiteName:    testSuite,
+				FailingTestCases: failedTcs,
+			})
+		}
+	}
+
+	return testSuites, nil
+}
+
+// Main function for the `show failures` subcommand.
+func showFailures(_ *cobra.Command, _ []string) error {
+	outputFormat, err := parseOutputFormatFlag()
+	if err != nil {
+		return err
+	}
+
+	// Parse the claim file into the claim scheme.
+	claimScheme, err := claim.Parse(claimFilePathFlag)
```

Comment:
Optional(could be another PR): Add a claim format check before trying to decode the claim file